sdk/core/azure-core/azure/core/credentials.py

```diff
@@ -111,3 +111,49 @@ def update(self, signature):
         if not isinstance(signature, six.string_types):
             raise TypeError("The signature used for updating must be a string.")
         self._signature = signature
+
+
+class AzureNamedKeyCredential(object):
+    """Credential type used for working with any service needing a named key that follows patterns
+    established by the other credential types.
+
+    :param str name: The name of the credential used to authenticate to an Azure service.
+    :param str key: The key used to authenticate to an Azure service.
+    :raises: TypeError
+    """
+    def __init__(self, name, key):
+        # type: (str, str) -> None
+        if not isinstance(name, six.string_types) or not isinstance(key, six.string_types):
+            raise TypeError("Both name and key must be Strings.")
+        self._name = name
+        self._key = key
+
+    @property
+    def name(self):
+        # type () -> str
+        """The value of the configured name.
+
+        :rtype: str
+        """
+        return self._name
+
+    @property
+    def key(self):
+        # type () -> str
+        """The value of the configured key.
+
+        :rtype: str
+        """
+        return self._key
+
+    def update(self, name, key):
```

Comment:
Do we have scenarios where you want to update/change the name of the key?

Reply:
I think one scenario could be that users want to use a new share access policy which has new name and key.

Reply:
Can user update key only?

Reply:
technically - they are allowed to provide the same name - but they must provide a name

Reply:
I think the most common scenario is the name is fixed after creation, but user may modify the key value?

Reply:
@yunhaoling  any thoughts on this?

If we know that users may normally not modify the name, I'll incline towards making name kwarg only

Reply:
actually on second thought, i think we should let this be - as much as i like kwargs, it doesn't seem like they'll grow later in this case.

Reply:
also, if the user thinks they wont need to change name for their use case, AzureKeyCredential serves their purpose

Reply:
...unless they need to initially specify the name?

Reply:
hmm, right - still doesn't change tht kwargs would not grow in this case - so will stick to it